iohk/iohk-ops.hs

```diff
@@ -289,4 +289,44 @@ runTemplate o@Options{..} Template{..} args = do
   echo ""
   echo $ "-- " <> (unsafeTextToLine $ configFilename) <> " is:"
   cmd o "cat" [configFilename]
-runTemplate _ _ _ = error "impossible"
+
+  -- nixops create:
+  Ops.create o config
+
+  -- generate dev-keys & ensure secrets exist:
+  when (tEnvironment == Development) $ do
+    let secrets = [ "static/github_token"
+                  , "static/id_buildfarm"
+                  , "static/datadog-api.secret"
+                  , "static/datadog-application.secret" ]
+    forM_ secrets touch
+    echo "Ensured secrets exist"
+
+    if (tGenerateKeys /= GenerateKeys)
+    then echo "Skipping key generation, due to user request"
+    else do
+      generateDevKeys o (clusterConfigurationKey config) "keys"
+      sh $ do
+        k <- Turtle.find ((prefix "keys/keys-testnet/rich/key") <> (suffix ".sk"))
+          "keys/keys-testnet/rich"
+        cp k $ "keys" </> Path.filename k
+  echo "Cluster deployment has been prepared."
+
+runNew _ _ _ = error "impossible"
+
+-- | Use 'cardano-keygen' to create keys for a develoment cluster.
+generateDevKeys :: Options -> ConfigurationKey -> Turtle.FilePath -> IO ()
```

Comment:
This should be named `generateGenesis` as a general case, then used for development clusters as needed.

Reply:
The genesis is generated by `cardano-node` during startup, so it's really just generation of keys compatible with a provided configuration key -- to the best of my understanding.

cc @gromakovsky

Reply:
Why is there parameter called `--genesis-out-dir`?

In that case we should give it a bit descriptive name, like `generateStakeKeys`.

Reply:
Probably because @pva701 is bad at naming things :)
In master we also have `dump-genesis-data --path PATH` command.
I can rename `generate-keys-by-spec` to `dump-genesis-secrets` (not keys, because it also dumps AVVM seeds) and its path argument to `--path`. I'd like to do it in master, because it's not something strictly necessary for 1.0.

Reply:
I'd still call this function `generateStakeKeys` as that's what it does, the fact that it's developer environment comes from it's usage, not definition.